app/code/Magento/Sales/Model/Order/Address/Renderer.php

```diff
@@ -62,15 +70,15 @@ public function __construct(
      */
     public function format(Address $address, $type)
     {
-        $storeId = $address->getOrder()->getStoreId();
-        $this->addressConfig->setStore($storeId);
+        $orderStore = $address->getOrder()->getStore();
+        $this->storeManager->setCurrentStore($orderStore);
```

Comment:
Changing the current store but not resetting results in lost context for following algorithms. So that's more of architectural issue then it is of concrete problem.